jokers/red_card.lua

```diff
@@ -0,0 +1,163 @@
+                            return true
+                        end
+                    }))
+                    SMODS.calculate_effect({
+                        message = localize{
+                            type = "variable",
+                            key = "a_folly_"..card.ability.extra.colour.."_card",
+                            vars = vars },
+                        colour = colours[card.ability.extra.colour].message_colour
+                    }, card)
+                    return true
+                end
+            }))
+        end
+
+        if context.joker_main then
+            if card.ability.extra.colour == "red" then
+                return {
+                    mult = card.ability.extra.mult * card.ability.extra.triggers,
+                }
+            end
+            if card.ability.extra.colour == "blue" then
+                return {
+                    chips = card.ability.extra.chips * card.ability.extra.triggers,
+                }
+            end
+        end
+    end,
+    loc_vars = function(self, info_queue, card)
+        --red by default
+        local vars =
+            { card.ability.extra.mult, card.ability.extra.mult * card.ability.extra.triggers }
+        if card.ability.extra.colour == "blue" then
+            vars =
+                { card.ability.extra.chips, card.ability.extra.chips * card.ability.extra.triggers }
+        end
+        if card.ability.extra.colour == "yellow" then
+            vars =
+                { card.ability.extra.money * card.ability.extra.triggers, card.ability.extra.money }
+        end
+        if card.ability.extra.colour == "green" then
+            vars = {
+                card.ability.extra.reroll * card.ability.extra.triggers,
+                card.ability.extra.reroll,
+            }
+        end
+        return {
+            key = colours[card.ability.extra.colour].loc_key,
+            vars = vars,
+        }
+    end,
+
+    switch_colour = function(self, card)
+        local prev = card.ability.extra.colour
+        local next = colours[card.ability.extra.colour].next
+        card.ability.extra.colour = next
+
+        if next == "green" or prev == "green" then
+            self.update_reroll_cost(card)
+        end
+    end,
+
+    update_reroll_cost = function(card)
+        local triggers = card.ability.extra.triggers
+        if card.ability.extra.colour ~= "green" then
+            triggers = triggers - 1
+        end
+        local reroll_cost = card.ability.extra.reroll * triggers
+        if card.ability.extra.colour == "green" then
+            G.GAME.round_resets.reroll_cost = G.GAME.round_resets.reroll_cost - reroll_cost
+            G.GAME.current_round.reroll_cost = G.GAME.current_round.reroll_cost - reroll_cost
+        else
+            G.GAME.round_resets.reroll_cost = G.GAME.round_resets.reroll_cost + reroll_cost
+            G.GAME.current_round.reroll_cost = G.GAME.current_round.reroll_cost + reroll_cost
```

Comment:
selling green card while you have free rerolls (chaos joker) causes you to lose that free reroll.
you probably need to call `calculate_reroll_cost(true)` afterwards

Reply:
also doesn't interact great with d6 tag